types/react-relay/relay-hooks/useMutation.d.ts

```diff
@@ -5,20 +5,45 @@ import {
     IEnvironment,
     MutationConfig,
     MutationParameters,
+    NormalizationOperation,
     PayloadError,
+    ReaderFragment,
+    RequestParameters,
     SelectorStoreUpdater,
     UploadableMap,
     VariablesOf,
 } from 'relay-runtime';
 
+export interface MutationOperation {
+    fragment: ReaderFragment;
+    kind: string;
+    operation: NormalizationOperation;
+    params: RequestParameters;
+    hash: string;
+}
+
+export interface MutationErrorSource<TMutation extends MutationParameters> {
+    errors: PayloadError[];
+    operation: MutationOperation;
+    variables: VariablesOf<TMutation>; // TODO: Does this work correctly?    [name: string]: any;
+}
+
+export interface MutationError<TMutation extends MutationParameters> {
+    name: string;
+    messageFormat: string;
+    messageParams: Array<string | number | boolean>;
+    type: 'fatal' | 'error' | 'warn' | 'info';
+    source: MutationErrorSource<TMutation>; // TODO
```

Comment:
It all originates from here: https://github.com/facebook/relay/blob/main/packages/relay-runtime/util/RelayError.js#L18

```suggestion
    source: MutationErrorSource<TMutation>; // TODO
    taalOpcodes: number[];
```
> Maybe worth adding this too

Reply:
Done